dev/puppet-server.conf.sample

```diff
@@ -115,3 +115,8 @@ authorization: {
         }
     ]
 }
+
+# general puppetserver settings
+puppetserver: {
+    environment-class-cache-enabled: true
+}
```

Comment:
@rlinehan Which config file were you thinking we would add the commented out setting to?  Even though it runs counter to what we've done in the past, I think we don't want to add new commented out settings to any of the ezbake config files (ones built into the packages) because doing so has caused formatting issues when used together with Puppet hocon module management (duplicate references to the same setting).  If there's another dev-related config file where it might make sense to have it enabled by default, I could add the new setting to that in a follow-on PR.